### ISSUE TITLE
Send inbound `TransactionsById` requests to the mempool storage service

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/zebra#1077]
**Is your feature request related to a problem? Please describe.**

The `zebra_network::Request::TransactionsById` variant should be handled by the inbound network service.

**Describe the solution you'd like**

- [ ] This page on the Bitcoin wiki https://en.bitcoin.it/wiki/Protocol_documentation#getdata suggests that transaction lookup is only allowed for mempool transactions, to prevent reliance on nodes having transaction indexes.  We should check what `zcashd` does.

- [ ] The inbound network service should query the mempool for those transactions and construct a response.

- [ ] The response should consist of the found transactions, followed by a list of NotFound hashes
  - [ ] `zcashd` ignores `NotFound`, but [its code says that SPV clients rely on `NotFound`](https://github.com/zcash/zcash/blob/e7b425298f6d9a54810cb7183f00be547e4d9415/src/main.cpp#L5632). Do SPV clients exist on the Zcash network?

**Additional context**

See: 
- https://developer.bitcoin.org/reference/p2p_networking.html#getdata 
- https://developer.bitcoin.org/reference/p2p_networking.html#tx 
- https://developer.bitcoin.org/reference/p2p_networking.html#notfound

Blocked on actually having a mempool.  Part of #889.

**Comment by oxarbitrage** [comment]
The `TransactionsByHash` was renamed to `TransactionsById` in https://github.com/ZcashFoundation/zebra/commit/c6082602561a15c0dfb37578a9b09b3cd8e017a4 so i changed the title and the contents of this ticket as it confused me to create a new request but it is not needed.

**Comment by teor2345** [comment]
> The `TransactionsByHash` was renamed to `TransactionsById` in [c608260](https://github.com/ZcashFoundation/zebra/commit/c6082602561a15c0dfb37578a9b09b3cd8e017a4) so i changed the title and the contents of this ticket as it confused me to create a new request but it is not needed.

Thanks!

Sorry I missed this ticket, I changed most of the names last week.

**Comment by oxarbitrage** [comment]
The reply to this message is actually 2 replies. One will have the transactions in the mempool we found but we also need to send a `notfound` message with a list of not found hashes.

This is what i think we can do and some questions i have:

1- Add code in https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/inbound.rs#L284 that will call https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/mempool.rs#L83 that will call https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/mempool/storage.rs#L100

2- That will get us a collection on what was found in the mempool. We can compare with the hashes in the request we have available here https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/inbound.rs#L284 and construct an additional response with not found hashes.

3- I am not sure on how we can reply with 2 messages in the inbound.rs as all the messages we have just reply with one. Should we wrap the 2 messages in a new response type and send that instead ?

**Comment by teor2345** [comment]
> The reply to this message is actually 2 replies. One will have the transactions in the mempool we found but we also need to send a `notfound` message with a list of not found hashes.

The minimal scope for this ticket is just the transactions.
(The ticket was opened before we did the mempool scope and design.)

It looks like `zcashd` and Zebra don't actually need `notfound` messages:
https://github.com/zcash/zcash/blob/e7b425298f6d9a54810cb7183f00be547e4d9415/src/main.cpp#L5632
https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/peer/connection.rs#L115

So can we open another ticket for `notfound`, and do it later if we have time?

**Comment by oxarbitrage** [comment]
Ok, cool, that makes this ticket a lot easier to code. Will send a PR.

**Comment by teor2345** [comment]
> It looks like `zcashd` and Zebra don't actually need `notfound` messages:
> https://github.com/zcash/zcash/blob/e7b425298f6d9a54810cb7183f00be547e4d9415/src/main.cpp#L5632
> https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/peer/connection.rs#L115
> 
> So can we open another ticket for `notfound`, and do it later if we have time?

I opened ticket #2726 for `notfound`.
